src/bms/unparse.rs

```diff
@@ -16,8 +126,49 @@ impl<T: KeyLayoutMapper> Bms<T> {
     #[must_use]
     pub fn unparse<'a>(&'a self) -> Vec<Token<'a>> {
         let mut tokens: Vec<Token<'a>> = Vec::new();
+        let mut needs_base62_token = false;
```

Comment:
To check whether base 62 is needed, how about to ask `ObjIdManager`s whether the ids in base 62 range are used? Checking with this boolean flag seems dangerous and annoying.